contracts/Company.sol

```diff
@@ -10,43 +12,44 @@ contract Company {
         ClassA
     }
 
-    address payable owner; //company admin
-    address employeeFund;
-    uint public currentSalary;
-    uint pension = 0;
-    salary_class class;
-    uint timeSinceClassChange;
-
-    //we will give address and initalSalary at time of deploying smart contract
-    //we are deploying contract when the employee is hired
-    constructor(
-        address _employeeFund,
-        uint _initSal,
-        salary_class _class
-    ) payable {
-        owner = payable(msg.sender);
-        employeeFund = _employeeFund;
-        currentSalary = _initSal;
-        class = _class;
-        timeSinceClassChange = block.timestamp;
+
+    struct Employee{
+        address account;
+        uint currentSalary;
+        salary_class class;
+        uint timeSinceClassChange;
+        uint experience;
     }
 
+    mapping(address=>Employee) public employeeList;
+    
+
+    constructor(address _fundContractAddress){
+        empFundContract = _fundContractAddress;
```

Comment:
We can also initialize the employeeList here in the constructor for setting up initial employees for convenience. 
Please let me know your thoughts